client/src/screens/events/Detail.js

```diff
@@ -71,6 +71,24 @@ class Detail extends Component {
     return mapMarkers;
   }
 
+  static getDerivedStateFromProps(newProps) {
```

Comment:
i'm not going to block this PR if you think it improves it but...
Why do we implement this at all? This should be able to be done in render with low CPU penalty. We should avoid storing state in the component unless absolutely necessary or we actually have a bad perf problem. Leave it for now - I'll have a go at sorting it out later once more critical stuff is done.

Reply:
If we hold it in state then the individual map makers can be notified of a state change if needed.

If we hold it in 'this' or generate it at render time we are pushing new markers every time.

Well thats my understanding and that was done in a PR long ago, unrelated to this specific change directly.

Reply:
![](https://cdn-images-1.medium.com/max/1600/1*jXzqMnnrXfePvfVglIYm5Q.jpeg)

Reply:
I don't think that's how it works. As long as the markers you're rendering are exactly the same (attributes/properties) there should be no need to re-init the map.